Update QuakeInjector to work with the new Quaddicted backend API

johnnycz opened the pull request:
Update QuakeInjector to work with the new Quaddicted backend API

**Major changes:**

- Works with the new Quaddicted backend API
- Size column added
- Both ratings columns removed

**Minor changes:**

- Sort by release date by default on startup
- Description box improved: bigger size, bigger screenshot, list all links
- Error handling improved

johnnycz committed:
use new Quaddicted JSON-based backend API

johnnycz committed:
remove rating columns, add size column

johnnycz commented:
Just found an issue where older maps don't have a start map defined so pressing "Play" results in an error. Will have a look at this.

johnnycz committed:
fix: start map should fallback to filename if none defined in metadata

johnnycz commented:
> Just found an issue where older maps don't have a start map defined so pressing "Play" results in an error. Will have a look at this.

Fixed

SpiritQuaddicted commented:
Sorry for being so late with feedback :(

This is so great, thank you!

Issues I found, I believe they are new to this branch:

- Size shows as 0 for many files, best round up/ceil instead.
- _Install random map_ seems not to work anymore
- log and download progress are not visible by default ![log and download progress are not visible by default](https://github.com/user-attachments/assets/e2a90988-6c32-42bc-ba8f-58875ce86f00)
- log is too narrow by default ![log is too narrow by default](https://github.com/user-attachments/assets/019e4582-b7e3-4ace-9ebb-ea0a89c42353)
- Canceling a download seems to end up with a unhandled exception

-----

Dependencies are an ugly subject at the moment. Sorry for not thinking of them and warning you. They are not finished at all...

> <li>Dependency references are identifiers like this: "ad_v1_80p1final", however this identifier is not present
>  *     in the actual dependency. Closest we get is filename: "ad_v1_80p1final.zip" which works fine, just a bit sketchy</li>

That way of specifying dependencies is meant to be deprecated at some point. There is a better-and-worse new way more like https://wiki.archlinux.org/title/PKGBUILD. E.g. `depends=('ad>=1.80')`, which should correspond to one or more packages with `provides='ad=1.80'` or similar.

The current handling should be ok for now though, so best not touch anything related to that. I would be very interested in thoughts on dependency and "provides" specification for other people than me, best on the forums not here.

johnnycz committed:
fix: Size shows as 0 for many files

johnnycz committed:
fix: info and log too narrow by default

johnnycz committed:
do not print exception for download cancellation

johnnycz commented:
Thanks for the great feedback!

> * Size shows as 0 for many files, best round up/ceil instead.

Fixed, added 1 decimal place.

> * _Install random map_ seems not to work anymore

Could not reproduce unfortunately. I wonder if this is map specific and what map is failing. If not then I'll need screenshot and log.

> * log and download progress are not visible by default
> * log is too narrow by default

Fixed by setting the default window size to be a bit larger than before.

> * Canceling a download seems to end up with a unhandled exception

I fixed canceling the map list download - now it won't print the very long stack traces. But if this was about something else a screenshot / log would be useful as canceling map install process does not produce any unhandled exceptions on my end.

SpiritQuaddicted commented:
Sorry I don't have more time on this and should invest available time in other issues :(((

If you want to fork or actively collaborate with others who have rights in this repo (I am too dumb to find the list), you would have my full endorsement and I would actively advertise it!

(But as I said, that API will most probably go away some time soon, and the JSON structure will probably get reduced again (it's messy... https://github.com/Quaddicted/quaddicted-data/issues/32))

johnnycz commented:
Makes sense, that's completely understandable. I've created a release candidate build and asked the community to help testing it: https://discuss.quaddicted.com/t/quake-injector-version-7-rc1-works-with-the-new-website/8079

I will process the community feedback on this update and ensure final official release exists that could then be officially hosted on Quaddicted.

I'd be happy to adapt things to any new API version once it is out. It'd be super useful if every version of the API had its own URL (eg. /v2/jsons or similar) and old API versions would keep working for at least a couple of months so that there's enough time for apps like Quake Injector and its users to transition to the newer version.

As for the future of the entire project itself and its ownership, I think it'd be best to discuss that somewhere else later, after this update is released officially.

SpiritQuaddicted commented:
Thanks so much!

Yes, the API should definitely get some nice versioned URLs. I hate myself for not thinking of that right from the beginning. I'll be in touch when it is almost ready so that we can prevent stupid mistakes (on my side) early on :)

SpiritQuaddicted commented:
Let's merge this, it is the new alpha07 :)

Thank you **so much**!
